pkg/kubelet/kubelet.go

```diff
@@ -2845,6 +2855,117 @@ func (kl *Kubelet) fastStatusUpdateOnce() {
 	}, 100*time.Millisecond, stopCh)
 }
 
+// handleMaxContainerCheckpoints goes through the list of checkpoints of a
+// container in a pod and deletes old checkpoints if there are more than
+// 'kl.maxContainerCheckpoints'.
+func (kl *Kubelet) handleMaxContainerCheckpoints(
+	podUID types.UID,
+	podFullName,
+	containerName string,
+	options *runtimeapi.CheckpointContainerRequest,
+) error {
+
+	if kl.maxContainerCheckpoints <= 0 {
+		// This cannot really happen as the kubelet config parser will
+		// error out early if this is <= 0. This additional error
+		// handling is mainly here to make sure tests are working.
+		return errors.New("maxContainerCheckpoints must be greater than 0")
+	}
+
+	// Lock while accessing the checkpoints on disk to avoid chaos
+	// if multiple threads would do it at the same time.
+	kl.maxContainerCheckpointsLock.Lock()
+	defer kl.maxContainerCheckpointsLock.Unlock()
+
+	checkpointDirectory, err := os.ReadDir(kl.getCheckpointsDir())
+	if err != nil {
+		return fmt.Errorf(
+			"error reading checkpoint archive directory %s: %w",
+			kl.getCheckpointsDir(),
+			err,
+		)
+	}
+
+	regexString := fmt.Sprintf(
+		`checkpoint-%s-%s-\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.*tar`,
+		podFullName,
+		containerName,
```

Comment:
Can you point me out to the code creating this file? Is time.RFC3339 used there? If so, can we somehow use that constant here and below in similar cases?

Reply:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L2868

Reply:
Thanks. I think it makes sense to use `time.Parse` instead of regex here, e.g. https://go.dev/play/p/dXygrHEpfQB

Reply:
Too parse the time? I am not interested in the time as it is part of `ReadDir()` like you pointed out. I just want something that matches the file name of that namespace/pod/container combination. The actual time in the file name is not important at this point.

Reply:
I think I get it. Let me try it out.

Reply:
I'm not sure I follow. Can you elaborate?

Reply:
The example in https://go.dev/play/p/agJV-ChICxo should make it clearer.